e2e/sign/sign.go

```diff
@@ -0,0 +1,180 @@
+
+package sign
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+)
+
+type ctx struct {
+	env             e2e.TestEnv
+	imgCache        string
+	keyringDir      string
+	passphraseInput []e2e.SingularityConsoleOp
+}
+
+const imgURL = "library://sylabs/tests/unsigned:1.0.0"
+const imgName = "testImage.sif"
+
+func (c *ctx) singularitySignHelpOption(t *testing.T) {
+	c.env.KeyringDir = c.keyringDir
+	c.env.RunSingularity(
+		t,
+		e2e.WithPrivileges(false),
+		e2e.WithCommand("sign"),
+		e2e.WithArgs("--help"),
+		e2e.ExpectExit(
+			0,
+			e2e.ExpectOutput(e2e.ContainMatch, "Attach a cryptographic signature to an image"),
+		),
+	)
+}
+
+func (c *ctx) prepareImage(t *testing.T) (string, func(*testing.T)) {
+	// Get a refresh unsigned image
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("failed to create temporary directory: %s", err)
+	}
+	imgPath := filepath.Join(tempDir, imgName)
+	e2e.PullImage(t, c.env, imgURL, imgPath)
+
+	return filepath.Join(tempDir, "testImage.sif"), func(t *testing.T) {
+		err := os.RemoveAll(tempDir)
+		if err != nil {
+			t.Fatalf("failed to delete temporary directory: %s", err)
+		}
+	}
+}
+
+func (c *ctx) singularitySignIDOption(t *testing.T) {
+	imgPath, cleanup := c.prepareImage(t)
+	defer cleanup(t)
+
+	cmdArgs := []string{"--id", "0", imgPath}
+	c.env.KeyringDir = c.keyringDir
+	c.env.ImgCacheDir = c.imgCache
+	c.env.RunSingularity(
+		t,
+		e2e.WithPrivileges(false),
+		e2e.WithCommand("sign"),
+		e2e.WithArgs(cmdArgs...),
+		e2e.ConsoleRun(c.passphraseInput...),
+		e2e.ExpectExit(
+			0,
+			e2e.ExpectOutput(e2e.ContainMatch, "Signature created and applied to "+imgPath),
+		),
+	)
+}
+
+func (c *ctx) singularitySignGroupIDOption(t *testing.T) {
+	imgPath, cleanup := c.prepareImage(t)
+	defer cleanup(t)
+
+	cmdArgs := []string{"--groupid", "0", imgPath}
+	c.env.KeyringDir = c.keyringDir
+	c.env.ImgCacheDir = c.imgCache
+	c.env.RunSingularity(
+		t,
+		e2e.WithPrivileges(false),
+		e2e.WithCommand("sign"),
+		e2e.WithArgs(cmdArgs...),
+		e2e.ConsoleRun(c.passphraseInput...),
+		e2e.ExpectExit(
+			0,
+			e2e.ExpectOutput(e2e.ContainMatch, "Signature created and applied to "+imgPath),
+		),
+	)
+}
+
+func (c *ctx) singularitySignKeyidxOption(t *testing.T) {
+	imgPath, cleanup := c.prepareImage(t)
+	defer cleanup(t)
+
+	cmdArgs := []string{"--keyidx", "0", imgPath}
+	c.env.KeyringDir = c.keyringDir
+	c.env.ImgCacheDir = c.imgCache
+	c.env.RunSingularity(
+		t,
+		e2e.WithPrivileges(false),
+		e2e.WithCommand("sign"),
+		e2e.WithArgs(cmdArgs...),
+		e2e.ConsoleRun(c.passphraseInput...),
+		e2e.ExpectExit(
+			0,
+			e2e.ExpectOutput(e2e.ContainMatch, "Signature created and applied to "+imgPath),
+		),
+	)
+}
+
+func (c *ctx) generateKeypair(t *testing.T) {
+	keyGenInput := []e2e.SingularityConsoleOp{
+		e2e.ConsoleSendLine("e2e sign test key"),
+		e2e.ConsoleSendLine("geoffroy@sylabs.io"),
```

Comment:
I would not use your own email here. Perhaps a `jdoe@sylabs.io`  or something else?